pkgs/node/src/chain.zig

```diff
@@ -369,22 +376,25 @@ pub const BeamChain = struct {
             block.slot,
         });
 
+        // Get parent state to calculate slots processed
+        const parent_state = self.states.get(signedBlock.message.block.parent_root) orelse return BlockProcessingError.MissingPreState;
         const post_state = if (blockInfo.postState) |post_state_ptr| post_state_ptr else computedstate: {
-            // 1. get parent state
-            const pre_state = self.states.get(block.parent_root) orelse return BlockProcessingError.MissingPreState;
+            // 1. get parent state (already retrieved above)
             const cpost_state = try self.allocator.create(types.BeamState);
-            try types.sszClone(self.allocator, types.BeamState, pre_state.*, cpost_state);
+            try types.sszClone(self.allocator, types.BeamState, parent_state.*, cpost_state);
 
             // 2. apply STF to get post state
             var validSignatures = true;
             stf.verify_signatures(signedBlock) catch {
                 validSignatures = false;
             };
+            const transition_timer = zeam_metrics.lean_state_transition_time_seconds.start();
```

Comment:
remove timer, and undo parent_state changes, timer is directly to be used inside STF functions